### PR TITLE
Upgrade gqlgen

### LINKED ISSUE #725: Upgrade gqlgen for UI API Layer
We use very old version of gqlgen (actually, we are locked on a specific commit). Now it's time to upgrade to latest version - 0.5.1. It has breaking changes, so upgrading won't be that easy.

### PR DESCRIPTION
**Pull request by pkosiec**
**Description**

Changes proposed in this pull request:
- Upgrade gqlgen to 0.5.1
- Move all gqlgen-related configuration to `config.yml` file
- Fix compatibility after breaking changes of gqlgen

**Related issue(s)**
Resolves #725

### CODE REVIEW

File: components/ui-api-layer/internal/gqlschema/schema.graphql [hunk 1]
```diff
@@ -462,20 +462,19 @@ type Query {
...
     connectorService(remoteEnvironment: String!): ConnectorService!
 
-    environments(remoteEnvironment: String): [Environment]!
-    deployments(environment: String!, excludeFunctions: Boolean): [Deployment]!
-    resourceQuotas(environment: String!): [ResourceQuota]!
-    resourceQuotasStatus(environment: String!): ResourceQuotasStatus!
+    environments(remoteEnvironment: String): [Environment!]!
+    deployments(environment: String!, excludeFunctions: Boolean): [Deployment!]!
+    resourceQuotas(environment: String!): [ResourceQuota!]!
+    resourceQuotaStatus(environment: String!): ResourceQuotasStatus!
```
Comment (michal-hudy) [inline components/ui-api-layer/internal/gqlschema/schema.graphql]: This is breaking change. Did you verify if it is not used? If it is used then please mark old query as deprecated.

File: components/ui-api-layer/internal/gqlschema/serviceinstance.go [hunk 2]
```diff
@@ -3,14 +3,14 @@ package gqlschema
...
+	Name                    string                 `json:"name"`
+	Environment             string                 `json:"environment"`
+	ServiceClassName        *string                `json:"serviceClassName"`
+	ServiceClassDisplayName string                 `json:"ServiceClassDisplayName"`
+	ServicePlanName         *string                `json:"servicePlanName"`
+	ServicePlanDisplayName  string                 `json:"servicePlanDisplayName"`
+	ServicePlanSpec         *JSON                  `json:"servicePlanSpec"`
+	CreationTimestamp       time.Time              `json:"creationTimestamp"`
+	Labels                  []string               `json:"labels"`
+	Status                  *ServiceInstanceStatus `json:"status"`
```
Comment (michal-hudy) [inline components/ui-api-layer/internal/gqlschema/serviceinstance.go]: I think `ServiceInstanceStatus` shouldn't be nullable. Maybe it should be fixed in another issue.

File: components/ui-api-layer/internal/domain/servicecatalog/instance_converter.go [hunk 3]
```diff
@@ -79,10 +79,15 @@ func (c *instanceConverter) GQLCreateInputToInstanceCreateParameters(in *gqlsche
 		parameterSchema = *in.ParameterSchema
 	}
 
+	var labels []string
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply components/ui-api-layer/internal/domain/servicecatalog/instance_converter.go]: yes

[the remark this answers is not in the document]
Comment (magicmatatjahu) [reply components/ui-api-layer/internal/domain/servicecatalog/instance_converter.go]: Ok, thanks for response. I understand.

[the remark this answers is not in the document]
Comment (pkosiec) [reply components/ui-api-layer/internal/domain/servicecatalog/instance_converter.go]: No problem. If you have any other questions, do not hesitate to ask.

File: components/ui-api-layer/internal/gqlschema/schema.graphql (hunk 1 above)
Comment (pkosiec) [reply components/ui-api-layer/internal/gqlschema/schema.graphql]: Sure, it was my fault during rebase. Already fixed

**Pull request merged**